### ISSUE TITLE
.transform output

### ISSUE THREAD

**Issue by mehrran** [MaartenGr/BERTopic#1005]
Hello @MaartenGr 

It looks like for the semi-supervised solution, the output of the model is different for the same document when we pass the exact training data in the same order or if we shuffle it. There are more -1 when we shuffle (or even pass the document to transform one by one)
let's say the code is:

umapModel = UMAP(n_neighbors=15, n_components=5, 
                  min_dist=0.0, metric='cosine', random_state=42)

topic_model = BERTopic(
    embedding_model=embedding_model,
    min_topic_size=3,
    umap_model=umapModel,
    calculate_probabilities=True)

topic_model.fit_transform(train_df['text'].to_list(),y=categories)

list1=train_df['text'].to_list()
list2 = random.sample(list1, len(list1))

topics1,probs1=topic_model.transform(list1)
topics2,probs2=topic_model.transform(list2)

d1 = {x:topics1.count(x) for x in topics1}
d2 = {x:topics2.count(x) for x in topics2}

print (d1)
print (d2)

The output is:

{5: 16, 4: 19, 3: 25, 35: 7, 10: 14, 0: 292, -1: 175, 75: 4, 13: 13, 9: 14, 57: 5, 65: 4, 2: 39, 14: 12, 1: 143, 82: 4, 54: 5, 60: 5, 7: 15, 43: 6, 45: 6, 6: 16, 11: 14, 18: 10, 20: 10, 8: 14, 53: 6, 86: 3, 36: 7, 40: 6, 62: 5, 25: 8, 84: 4, 50: 6, 74: 4, 49: 6, 12: 13, 19: 10, 59: 5, 27: 8, 83: 4, 78: 4, 24: 9, 89: 3, 68: 4, 29: 8, 70: 4, 32: 7, 28: 8, 56: 5, 38: 6, 52: 6, 39: 6, 23: 9, 90: 3, 79: 4, 73: 4, 17: 11, 15: 12, 81: 4, 64: 5, 55: 5, 37: 7, 66: 4, 77: 4, 22: 9, 76: 4, 47: 6, 71: 4, 41: 6, 44: 6, 51: 6, 69: 4, 48: 6, 87: 3, 88: 3, 80: 4, 67: 4, 21: 9, 61: 5, 42: 6, 46: 6, 31: 8, 33: 7, 58: 5, 63: 5, 26: 8, 30: 8, 85: 3, 34: 7, 16: 12, 72: 4}

{2: 13, 6: 6, -1: 598, 1: 143, 0: 292, 84: 4, 69: 3, 14: 5, 44: 7, 33: 7, 83: 4, 37: 12, 32: 9, 10: 15, 20: 2, 8: 4, 28: 7, 55: 1, 86: 1, 16: 12, 57: 5, 47: 5, 74: 4, 5: 8, 7: 16, 42: 6, 3: 2, 25: 4, 9: 1, 68: 3, 41: 2, 30: 6, 72: 4, 34: 5, 22: 5, 77: 4, 12: 1, 73: 3, 13: 2, 18: 3, 21: 3, 71: 4, 56: 2, 4: 5, 65: 4, 60: 6, 61: 2, 66: 1, 58: 2, 45: 2, 75: 3, 50: 1, 62: 4, 46: 1, 53: 1, 40: 3, 36: 1, 48: 1, 51: 1, 88: 1}

How come the number of -1s are too high in the second output? or some of the clusters not getting any member, or only 1. According to the fact that it was the training data and in the first list (which is in the same order during the training) the distribution is same as the fitting time)

**Comment by MaartenGr** [comment]
I believe this is a result of how HDBSCAN works. It approximates a prediction that tends to differ from its initial training. Moreover, I believe it hashes the original input so that when you apply `.transform` on the original input, in the original order, it will find the original predictions and use those.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
